bulk_index.py

```diff
@@ -0,0 +1,106 @@
+"""Use this to populate a search index in bulk or prefetch metadata."""
+
+import os
+import click
+from search.factory import create_ui_web_app
+from search.agent import MetadataRecordProcessor, DocumentFailed, \
+    IndexingFailed
+
+app = create_ui_web_app()
+app.app_context().push()
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
+
+
+@app.cli.command()
+@click.option('--paper_id', '-p',
+              help='Index specified paper id')
+@click.option('--id_list', '-l',
+              help="Index paper IDs in a file (one ID per line)")
+@click.option('--alt_cache_dir', '-c',
+              help="Specify alternate cache directory for document metadata")
+@click.option('--prefetch_metadata', '-m', is_flag=True,
+              help="Prefetch latest version document metadata, don't index")
+@click.option('--no_index', '-n', is_flag=True,
+              help="Don't index; use with --prefetch_metadata")
+def populate(paper_id, id_list, alt_cache_dir, prefetch_metadata, no_index):
+    """Populate the search index with some test data."""
+    cache_dir = os.path.abspath('tests/data/temp')
+    # Create default cache directory if it doesn't exist
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir)
+
+    if alt_cache_dir:
+        if not (os.path.exists and os.access(alt_cache_dir, os.W_OK)):
+            click.echo(
+                f'Path does not exist or is not writable: {alt_cache_dir}')
+            return
+        cache_dir = alt_cache_dir
+
+    PAPERS_PER_BULK = 500
+    MAX_ERRORS = int(os.environ.get('MAX_ERRORS', 5))
+
+    processor = MetadataRecordProcessor()
+    processor.init_cache(cache_dir)
+
+    error_count = 0
+    index_count = 0
+
+    if paper_id:    # Index a single paper.
+        TO_INDEX = [paper_id]
+    elif id_list:   # Index a list of papers.
+        if not os.path.exists(id_list):
+            click.echo('Path does not exist: %s' % id_list)
+            return
+        with open(id_list) as f:
+            TO_INDEX = [ident.strip() for ident in f.read().split()]
+
+    docs = []
+    for arxiv_id in TO_INDEX:
+        if 'v' in arxiv_id:
```

Comment:
There are a couple of edge cases were this won't work (e.g. solv-int/9806007). I'll push a fix for this script and populate_test_metadata.py shortly.

Reply:
Fix in PR #106